apps/studio/components/interfaces/Settings/Logs/LogsQueryPanel.tsx

```diff
@@ -44,86 +48,171 @@ export interface LogsQueryPanelProps {
           <div className="flex items-center gap-2">
-            <DropdownMenu>
-              <DropdownMenuTrigger asChild>
-                <Button type="default" iconRight={<IconChevronDown />}>
-                  Insert source
-                </Button>
-              </DropdownMenuTrigger>
-              <DropdownMenuContent side="bottom" align="start">
-                {logsTableNames
-                  .sort((a, b) => a.localeCompare(b))
-                  .map((source) => (
-                    <DropdownMenuItem key={source} onClick={() => onSelectSource(source)}>
-                      <div className="flex flex-col gap-1">
-                        <span className="font-mono font-bold">{source}</span>
-                        <span className="text-foreground-light">
-                          {LOGS_SOURCE_DESCRIPTION[source]}
+            {warehouseEnabled && (
+              <DropdownMenu>
+                <DropdownMenuTrigger asChild>
+                  <Button type="default" iconRight={<ChevronDown />}>
+                    Data source <span className="ml-2 font-mono opacity-50">{dataSource}</span>
+                  </Button>
+                </DropdownMenuTrigger>
+                <DropdownMenuContent side="bottom" align="start">
+                  <DropdownMenuItem onClick={() => onDataSourceChange('logs')}>
+                    <DropdownMenuItemContent name="Logs" desc="Logs for all Supabase products" />
+                  </DropdownMenuItem>
+
+                  <DropdownMenuItem onClick={() => onDataSourceChange('warehouse')}>
+                    <DropdownMenuItemContent
+                      name={
+                        <span>
+                          Warehouse <Badge variant="warning">NEW</Badge>
                         </span>
-                      </div>
-                    </DropdownMenuItem>
-                  ))}
-              </DropdownMenuContent>
-            </DropdownMenu>
+                      }
+                      desc="Query your data warehouse collections"
+                    />
+                  </DropdownMenuItem>
+                </DropdownMenuContent>
+              </DropdownMenu>
+            )}
 
-            <DropdownMenu>
-              <DropdownMenuTrigger asChild>
-                <Button type="default" iconRight={<IconChevronDown />}>
-                  Templates
-                </Button>
-              </DropdownMenuTrigger>
-              <DropdownMenuContent side="bottom" align="start">
-                {templates
-                  .sort((a, b) => a.label!.localeCompare(b.label!))
-                  .map((template) => (
+            {dataSource === 'warehouse' && (
+              <DropdownMenu>
+                <DropdownMenuTrigger asChild>
+                  <Button type="default" iconRight={<ChevronDown />}>
+                    Templates
+                  </Button>
+                </DropdownMenuTrigger>
+                <DropdownMenuContent>
```

Comment:
We should probably add a scroll area for this one when theres > n collections (can refer to ProjectDropdown)
![image](https://github.com/supabase/supabase/assets/19742402/31b332c0-8d8b-4eec-961d-ec688aaf3d60)

Reply:
also lets do align="start", consistent with the logs as data source